api/python/provisioner/salt.py

```diff
@@ -1143,15 +1143,22 @@ def _value(v):
 def states_apply(
     states: List[Union[str, State]], targets=ALL_MINIONS, **kwargs
 ):
-    # TODO multiple states at once
-    ret = {}
-    for state in states:
-        state = State(state)
-        res = function_run(
-            'state.apply', fun_args=[state.name], targets=targets, **kwargs
-        )
-        ret[state.name] = res
+    try:
+        # TODO multiple states at once
+        ret = {}
+        for state in states:
+            state = State(state)
+            res = function_run(
+                'state.apply', fun_args=[state.name], targets=targets, **kwargs
+            )
+            ret[state.name] = res
 
+        return ret
+    except Exception:
+        logger.exception(
```

Comment:
That level is too low to print traces. Outer scopes will likely do that as well and it would be an overhead for logs. Please revert that change.